resources/fetchers/network_fetcher.go

```diff
@@ -91,4 +79,36 @@ func (r NetworkResource) GetMetadata() (fetching.ResourceMetadata, error) {
 		Name:    r.GetResourceName(),
 	}, nil
 }
+
 func (r NetworkResource) GetElasticCommonData() any { return nil }
+
+func (f NetworkFetcher) aggregateResources(ctx context.Context, client ec2.ElasticCompute) ([]awslib.AwsResource, error) {
+	var resources []awslib.AwsResource
+	nacl, err := client.DescribeNetworkAcl(ctx)
+	if err != nil {
+		f.log.Errorf("failed to describe network acl: %v", err)
+	}
+	resources = appendList(resources, nacl)
+
+	securityGroups, err := client.DescribeSecurityGroups(ctx)
+	if err != nil {
+		f.log.Errorf("failed to describe security groups: %v", err)
+	}
+	resources = appendList(resources, securityGroups)
+
+	vpcs, err := client.DescribeVPCs(ctx)
+	if err != nil {
+		f.log.Errorf("failed to describe vpcs: %v", err)
+	}
+	resources = appendList(resources, vpcs)
+
+	return resources, nil
+}
+
+func appendList(list []awslib.AwsResource, resources []awslib.AwsResource) []awslib.AwsResource {
```

Comment:
Why this is needed?
append to nil is not harmful
```go
	r1 := []*Data{{}, {}} // slice with two items
	var r2 []*Data // zero value of slice is nil
	r3 := append(r1, r2...)
	fmt.Println(r2 == nil)
	fmt.Print(len(r3), cap(r3))
```

Will output
```bash
true
2 2
```

Reply:
reverted